src/Uno.UI/UI/Xaml/FrameworkElementExtensions.cs

```diff
@@ -324,6 +324,48 @@ internal static bool TrySetPadding(this IFrameworkElement frameworkElement, Thic
 			return false;
 		}
 
+		internal static bool TryGetActualBorderThickness(this IFrameworkElement frameworkElement, out Thickness borderThickness)
+		{
+			if (__LinkerHints.Is_Windows_UI_Xaml_Controls_Grid_Available && frameworkElement is Grid g)
+			{
+				borderThickness = g.BorderThickness;
+				return true;
+			}
+
+			if (__LinkerHints.Is_Windows_UI_Xaml_Controls_StackPanel_Available && frameworkElement is StackPanel sp)
+			{
+				borderThickness = sp.BorderThickness;
+				return true;
+			}
+
+			if (__LinkerHints.Is_Windows_UI_Xaml_Controls_RelativePanel_Available && frameworkElement is RelativePanel rp)
+			{
+				borderThickness = rp.BorderThickness;
+				return true;
+			}
+
+			if (__LinkerHints.Is_Microsoft_UI_Xaml_Controls_LayoutPanel_Available && frameworkElement is Microsoft.UI.Xaml.Controls.LayoutPanel lp)
+			{
+				borderThickness = lp.BorderThickness;
+				return true;
+			}
```

Comment:
Can all these be replaced by the base `Panel` and use https://github.com/unoplatform/uno/blob/0190ea92ffc138e1cc3259e59fdf2074cbc16465/src/Uno.UI/UI/Xaml/Controls/Panel/Panel.cs#L113 ?

Reply:
Good idea, that simplifies things a lot